gcloud/monitoring/timeseries.py

```diff
@@ -0,0 +1,152 @@
+  * Writing time series.
+  * Natural representation of distribution values.
+
+.. _Google Monitoring API (V3):
+    https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries
+"""
+
+import collections
+
+from gcloud.monitoring.metric import Metric
+from gcloud.monitoring.resource import Resource
+
+
+class TimeSeries(collections.namedtuple(
+        'TimeSeries', 'metric resource metric_kind value_type points')):
+    """A single time series of metric values.
+
+    :type metric: :class:`~gcloud.monitoring.metric.Metric`
+    :param metric: A metric object.
+
+    :type resource: :class:`~gcloud.monitoring.resource.Resource`
+    :param resource: A resource object.
+
+    :type metric_kind: string
+    :param metric_kind:
+        The kind of measurement: :data:`MetricKind.GAUGE`,
+        :data:`MetricKind.DELTA`, or :data:`MetricKind.CUMULATIVE`.
+        See :class:`~gcloud.monitoring.metric.MetricKind`.
+
+    :type value_type: string
+    :param value_type:
+        The value type of the metric: :data:`ValueType.BOOL`,
+        :data:`ValueType.INT64`, :data:`ValueType.DOUBLE`,
+        :data:`ValueType.STRING`, :data:`ValueType.DISTRIBUTION`,
+        or :data:`ValueType.MONEY`.
+        See :class:`~gcloud.monitoring.metric.ValueType`.
+
+    :type points: list of :class:`Point`
+    :param points: A list of point objects.
+    """
+
+    @property
+    def labels(self):
+        """A single dictionary with values for all the labels.
+
+        This combines ``resource.labels`` and ``metric.labels`` and also
+        adds ``"resource_type"``.
+        """
+        # pylint: disable=attribute-defined-outside-init
+        try:
+            return self._labels
+        except AttributeError:
+            labels = {'resource_type': self.resource.type}
+            labels.update(self.resource.labels)
+            labels.update(self.metric.labels)
+            self._labels = labels
+            return self._labels
+
+    def header(self, points=None):
+        """Copy everything but the point data.
+
+        :type points: list of :class:`Point`, or None
+        :param points: An optional point list.
+
+        :rtype: :class:`TimeSeries`
+        :returns: The new time series object.
+        """
+        points = list(points) if points else []
+        return self._replace(points=points)
+
+    @classmethod
+    def _from_dict(cls, info):
+        """Construct a time series from the parsed JSON representation.
+
+        :type info: dict
+        :param info:
+            A ``dict`` parsed from the JSON wire-format representation.
+
+        :rtype: :class:`TimeSeries`
+        :returns: A time series object.
+        """
+        metric = Metric._from_dict(info['metric'])
+        resource = Resource._from_dict(info['resource'])
+        metric_kind = info['metricKind']
+        value_type = info['valueType']
+        points = [Point._from_dict(p) for p in info.get('points', [])]
```

Comment:
Missed this in https://github.com/GoogleCloudPlatform/gcloud-python/pull/1691/commits/41e8ff20c658ce1d29574a1770f20d11dc266188